onnxruntime/core/providers/partitioning_utils.cc

```diff
@@ -0,0 +1,324 @@
+      // if the node input was not produced by this subgraph, add it to the subgraph inputs.
+      if (node_outputs.count(input) == 0) {
+        if (subgraph_inputs.count(input) == 0) {
+          subgraph_inputs.insert(input);
+          ordered_subgraph_inputs.push_back(input);
+        }
+      }
+    }
+
+    const auto& output_defs = node->OutputDefs();
+    for (const auto* output_def : output_defs) {
+      node_outputs.insert(output_def);
+      // if output is overall graph output we need to produce it.
+      if (graph_outputs.count(output_def) != 0) {
+        ordered_subgraph_outputs.push_back(output_def);
+      }
+    }
+
+    // if output connects to a node not in this subgraph we need to add it
+    // unless it was already added as an overall graph output,
+    for (auto it = node->OutputEdgesBegin(), end = node->OutputEdgesEnd(); it != end; ++it) {
+      if (node_set.count(&it->GetNode()) == 0) {
+        const auto* output_def = output_defs[it->GetSrcArgIndex()];
+        if (subgraph_outputs.count(output_def) == 0 && graph_outputs.count(output_def) == 0) {
+          subgraph_outputs.insert(output_def);
+          ordered_subgraph_outputs.push_back(output_def);
+        }
+      }
+    }
+  }
+
+  // Assign inputs and outputs to subgraph's meta_def
+  auto meta_def = std::make_unique<::onnxruntime::IndexedSubGraph::MetaDef>();
+  meta_def->name = generate_metadef_name();
+  meta_def->domain = execution_provider_name;
+  meta_def->since_version = 1;
+  meta_def->status = ONNX_NAMESPACE::EXPERIMENTAL;
+
+  for (const auto& input : ordered_subgraph_inputs) {
+    meta_def->inputs.push_back(input->Name());
+  }
+
+  for (const auto& output : ordered_subgraph_outputs) {
+    meta_def->outputs.push_back(output->Name());
+  }
+
+  sub_graph->SetMetaDef(std::move(meta_def));
+
+  return std::make_unique<ComputeCapability>(std::move(sub_graph));
+}
+
+std::vector<std::unique_ptr<ComputeCapability>>
+CreateSupportedPartitions(const GraphViewer& graph_viewer,
+                          const std::unordered_set<const Node*>& supported_nodes,
+                          const std::unordered_set<std::string>& stop_ops,
+                          const std::function<std::string()>& generate_metadef_name,
+                          const std::string& execution_provider_name,
+                          bool debug_output) {
+  // find any nodes we need to exclude
+  std::unordered_set<const Node*> excluded_nodes = CreateExcludedNodeSet(graph_viewer, stop_ops);
+
+#ifndef NDEBUG
+  auto node_str = [](const Node& node) {
+    std::ostringstream oss;
+    oss << node.Index() << " '" << node.Name() << "'(" << node.OpType() << ")";
+    return oss.str();
+  };
+
+  auto group_str = [&node_str](const std::vector<const Node*>& group) {
+    const Node& start_node = *group.front();
+    const Node& end_node = *group.back();
+    std::ostringstream oss;
+    oss << node_str(start_node) << " to " << node_str(end_node) << "\n";
+    return oss.str();
```

Comment:
This is already a debug message, 
Can we add all the nodes in the group? Or at lease node index?

Reply:
Not sure what you mean by it already being a debug message. Can you clarify?

All nodes in the group is not useful for a real model, as there may be hundreds of nodes. 

You currently get a message when a new partitions starts with details of the node that caused the partition to end (node index, name and optype). You also get that info for the start and end nodes in the partition. 

I started with a full list when checking real models but reduced it to that amount info when the output was overwhelming (e.g. thousands of nodes in mobile_bert). Primarily the optype that breaks the partition is the most relevant info though.

Reply:
Fair enough, I guess it may be too much information for huge model such as mobile_bert